### PR TITLE
[MDS-5686] Remove crash details in error responses in CORE-API

### PR DESCRIPTION
**Pull request by isuru-aot**
## Objective 

[MDS-5686](https://bcmines.atlassian.net/browse/MDS-5686)

_Why are you making this change? Provide a short explanation and/or screenshots_

- Added trace id to error responses in core-api
- Added the trace id to notification message in MS and Core
- Added the trace id to the error email 
   - Added a link to Kibana logs which with a filter to log the corresponding error messages
   - Introduced two separate email template for MS and Core

Note: In addition to the task, removed the unnecessary Custom Exception in `mines/exceptions/mine_exceptions.py` and update the refences with appropriate `messages` and `status codes`

### CODE REVIEW

File: services/core-api/app/api/mines/explosives_permit/resources/explosives_permit.py [hunk 1]
```diff
@@ -221,8 +220,7 @@ def put(self, mine_guid, explosives_permit_guid):
             explosives_permit.save()
         except Exception as e:
             current_app.logger.error(e)
-            raise MineException("Oops!, Something went wrong while updaitng the explosive permit information",
-                                detailed_error = e)
+            raise MineException("Something went wrong while updaitng the explosive permit information")
```
Comment (taraepp) [inline services/core-api/app/api/mines/explosives_permit/resources/explosives_permit.py]: typo on updating.
I do wonder a bit about using the "mine exception" type for all these- I do see that we also have `ExplosivesPermitDocumentException`, `ExplosivesPermitExeption` (some spelling errors there 😓 ) but also not sure that we need a different exception type for every object type when we're setting a custom error message in every instance.

Comment (isuru-aot) [reply services/core-api/app/api/mines/explosives_permit/resources/explosives_permit.py]: Removed unnecessary exceptions and used only `MineException` with appropriate message and status codes.

File: services/core-api/app/config.py [hunk 2]
```diff
@@ -207,6 +207,9 @@ def JWT_ROLE_CALLBACK(jwt_dict):
     # Enable flag caching and evalutation. If set to True, FLAGSMITH_KEY must be set to a server side FLAGSMITH_KEY
     FLAGSMITH_ENABLE_LOCAL_EVALUTION=os.environ.get('FLAGSMITH_ENABLE_LOCAL_EVALUTION', 'false') == 'true'
 
+    # Kibana
+    KIBANA_BASE_URL = os.environ.get('KIBANA_BASE_URL', 'https://kibana-openshift-logging.apps.silver.devops.gov.bc.ca')
```
Comment (henryoforeh-dev) [inline services/core-api/app/config.py:211]: For this environment variable a gitops PR need to be created too.

**Review by henryoforeh-dev — APPROVED** [review]

File: services/core-api/app/api/mines/explosives_permit/resources/explosives_permit.py (hunk 1 above)
Comment (taraepp) [reply services/core-api/app/api/mines/explosives_permit/resources/explosives_permit.py]: I know the typo is pre-existing, but should still change 'updaitng' to 'updating'

Comment (isuru-aot) [reply services/core-api/app/api/mines/explosives_permit/resources/explosives_permit.py]: @taraepp Well, that's updated already. Please see all the changes.

Comment (taraepp) [reply services/core-api/app/api/mines/explosives_permit/resources/explosives_permit.py]: So it is! Must have been looking at it wrong. 🤷‍♀️

**Review by taraepp — APPROVED** [review]

**Pull request merged**